lib/grandpa/neighbor_tracker.go

```diff
@@ -0,0 +1,121 @@
+package grandpa
+
+import (
+	"fmt"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"time"
+)
+
+type neighborData struct {
+	peer        peer.ID
+	neighborMsg *NeighbourPacketV1
+}
+
+type neighborState struct {
+	setID uint64
+	round uint64
+	//highestFinalized uint32 not sure if i need this or not
+}
+
+type NeighborTracker struct {
+	grandpa *Service
+
+	peerview         map[peer.ID]neighborState
+	currentSetID     uint64
+	currentRound     uint64
+	highestFinalized uint32
+
+	finalizationCha chan *types.FinalisationInfo
+	neighborMsgChan chan neighborData
+	stoppedNeighbor chan struct{}
+}
+
+func NewNeighborTracker(grandpa *Service, neighborChan chan neighborData) *NeighborTracker {
+	return &NeighborTracker{
+		grandpa:         grandpa,
+		peerview:        make(map[peer.ID]neighborState),
+		finalizationCha: grandpa.blockState.GetFinalisedNotifierChannel(),
+		neighborMsgChan: neighborChan,
+		stoppedNeighbor: make(chan struct{}),
+	}
+}
+
+func (nt *NeighborTracker) Start() {
+	go nt.run()
+}
+
+func (nt *NeighborTracker) Stop() {
+	nt.grandpa.blockState.FreeFinalisedNotifierChannel(nt.finalizationCha)
+	nt.stoppedNeighbor <- struct{}{}
+	close(nt.neighborMsgChan)
```

Comment:
I don't think we should close this channel here as the `*NeighborTracker` is not the writer of this channel, so I would leave the closing to the writer